pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -270,6 +282,23 @@ func (r *MachineSetSyncReconciler) reconcileMAPIMachineSetToCAPIMachineSet(ctx c
 	newCAPIMachineSet.SetNamespace(r.CAPINamespace)
 	newCAPIMachineSet.Spec.Template.Spec.InfrastructureRef.Namespace = r.CAPINamespace
 
+	convertedCAPIMachineSetOwnerReferences, err := r.convertMAPIMachineSetOwnerReferencesToCAPI(ctx, mapiMachineSet)
```

Comment:
Could we return something that is also usable later for newCAPIInfraMachineTemplate instead of only for newCAPIMachineSet?